### PR TITLE
add check for verification of unshield/burns

### PR DESCRIPTION
**Pull request by jowparks**
## Summary
Verification of burns, with mocked `TxResult`
## Testing Plan
tests pass

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```

### CODE REVIEW

**Review by patnir — APPROVED** [review]
Looks good, some minor comments/ questions

File: ironfish/src/consensus/verifier.ts [hunk 1]
```diff
@@ -648,21 +648,21 @@ export class Verifier {
...
-    //     return { valid: false, reason: VerificationResultReason.EVM_ASSET_MISMATCH }
-    //   }
-    // }
-
-    // TODO(jwp): verify shielding/mints balance
 
+    const mintResult = Verifier.verifyEvmMints(transaction, result)
+    if (!mintResult.valid) {
+      return mintResult
+    }
```
[the remark this answers is not in the document]
Comment (jowparks) [reply ironfish/src/consensus/verifier.ts:655]: No we can exit early if one fails, will only return if mintResult is failure.

File: ironfish/src/consensus/verifier.ts [hunk 2]
```diff
@@ -689,6 +689,36 @@ export class Verifier {
...
   }
 
+  static verifyEvmBurns(transaction: Transaction, result: EvmResult): VerificationResult {
+    // TODO(jwp): handle native asset as output balance (will require decrypting notes and comparing)
+    const assetBalanceDeltas = new AssetBalances()
+
+    for (const event of result.events) {
+      if (event.assetId.equals(Asset.nativeId())) {
+        continue
+      }
```
[the remark this answers is not in the document]
Comment (jowparks) [reply ironfish/src/consensus/verifier.ts:699]: Yea thats a good idea, shouldn't be possible so we can except.

File: ironfish/src/consensus/verifier.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jowparks) [reply ironfish/src/consensus/verifier.ts:655]: Oh I misread your question, hmm good question. Technically you could do both in the same transaction if we solve the signing problem/fee problem.

**Pull request merged**